notebooks/ullyses-custom-stis.ipynb

```diff
@@ -0,0 +1,686 @@
+    "\n",
+    "All YAML files used by the ULLYSES team are available in the [`ullyses_utils` package](https://github.com/spacetelescope/ullyses-utils/tree/main/src/ullyses_utils/data/stis_configs). Each STIS exposure was manually inspected by the ULLYSES team to account for any irregularities in the data. \n",
+    "\n",
+    "Let's take a look at the YAML files for SZ-76 now."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Find your local installation of the ullyses_utils package\n",
+    "utils_dir = ullyses_utils.__path__[0]\n",
+    "# Get the path to the STIS YAML files\n",
+    "yamls = glob.glob(os.path.join(utils_dir, f\"data/stis_configs/*sz-76*.yaml\"))\n",
+    "yamls.sort()\n",
+    "pprint.pprint(yamls)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There is one YAML file for each observed grating. Let's take a look at each one."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for yaml in yamls:\n",
+    "    # Use the ULLYSES convenience function to read each YAML file\n",
+    "    config = read_config(yaml)\n",
+    "    print(f\"{os.path.basename(yaml)}:\")\n",
+    "    pprint.pprint(config)\n",
+    "    print(\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each YAML file includes information about special calibration parameters needed to obtain optimized 1D spectra. Note the extra defringing parameters for the G750L data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=\"sec4\"></a>\n",
+    "## Reprocessing data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To rectify the data, we must reprocess it using the custom ULLYSES wrapper around CalSTIS. This wrapper can perform any number of corrections as specified in an input YAML file. This includes: \n",
+    "- Removing the fringe pattern. Using the [`stistools` defringing](https://stistools.readthedocs.io/en/latest/defringe_guide.html) package, the amplitude, shift, and position of the fringes are determined for each science spectrum. This is done by processing the contemporaneous fringe flat taken right before or after each G750L or G750M science exposure.\n",
+    "- Manually calculating the number of pixels with high dark rate (DQ=16), using a modified threshold.\n",
+    "- Performing a custom extraction to one or more targets in the long slit. This is done e.g. when the target is especially faint and requires a smaller extraction box, when the target is miscentered in the slit, or there is a companion source in the slit.\n",
+    "\n",
+    "Let's reprocess the SZ-76 data by using the YAML files as input to the ULLYSES code."
```

Comment:
Make a quick note here that there are plots produced by the function after the code finishes running, but we'll look at those in more detail in the sections below.